object.c

```diff
@@ -520,7 +520,9 @@ rb_obj_size(VALUE self, VALUE args, VALUE obj)
  *
  *  Yields <i>obj</i> and returns the result.
  *
- *	'my string'.yield_self {|s|s.upcase} #=> "MY STRING"
+ *  require "ostruct"
+ *  user = OpenStruct.new(firstname: "John", lastname: "Doe")
+ *  user.yield_self { |u| "#{u.firstname} #{u.lastname}" } #=> "John Doe" 
```

Comment:
The example in the comment by @bogdanvlviv  presents an interesting example using chaining:
```ruby
if application_default_user.male?
  "Mr. #{application_default_user.name}"
else
  "Ms. #{application_default_user.name}"
end
# But now you can:
application_default_user.yield_self { |u| u.male? ? "Mr. #{u.name}" : "Ms. #{u.name}" }
```